plugins/modules/proxmox_pool.py

```diff
@@ -0,0 +1,180 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2023, Sergei Antipov (UnderGreen) <greendayonfire@gmail.com>
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = r"""
+---
+module: proxmox_pool
+short_description: Pool management for Proxmox VE cluster
+description:
+  - Create or delete a pool for Proxmox VE clusters.
+  - For pool members management please consult proxmox_pool_member module.
+version_added: 7.1.0
+author: "Sergei Antipov (@UnderGreen) <greendayonfire@gmail.com>"
+attributes:
+  check_mode:
+    support: full
+  diff_mode:
+    support: none
+options:
+  poolid:
+    description:
+      - The pool id.
+    type: str
+    aliases: [ "name" ]
+    required: true
+  state:
+    description:
+     - Indicate desired state of the pool.
+     - The pool must be empty prior deleting it with I(state=absent).
```

Comment:
Since community.general 7.1.0 we can (finally) use semantic markup (https://docs.ansible.com/ansible/devel/dev_guide/developing_modules_documenting.html#semantic-markup-within-module-documentation):
```suggestion
     - The pool must be empty prior deleting it with O(state=absent).
```